plasmapy/utils/decorators/tests/test_validators.py

```diff
@@ -29,40 +37,22 @@ class :class:`~plasmapy.utils.decorators.validators.ValidateQuantities`.
     )  # type: Dict[str, Any]
     check_defaults = {**unit_check_defaults, **value_check_defaults}
 
-    @staticmethod
-    def foo(x):
-        return x
-
-    @staticmethod
-    def foo_anno(x: u.cm):
-        return x
-
-    def test_inheritance(self):
-        assert issubclass(ValidateQuantities, CheckUnits)
-        assert issubclass(ValidateQuantities, CheckValues)
-
-    def test_vq_method__get_validations(self):
-        # method must exist
-        assert hasattr(ValidateQuantities, "validations")
-        assert hasattr(ValidateQuantities, "_get_validations")
-
-        # setup default validations
-        default_validations = {
-            **self.check_defaults.copy(),
-            "units": [self.check_defaults["units"]],
-        }
-
-        # setup test cases
-        # 'setup' = arguments for `_get_validations`
-        # 'output' = expected return from `_get_validations`
-        # 'raises' = if `_get_validations` raises an Exception
-        # 'warns' = if `_get_validations` issues a warning
-        #
-        _cases = [
+    @pytest.mark.parametrize("_class", [CheckUnits, CheckValues])
+    def test_inheritance(self, _class):
+        assert issubclass(ValidateQuantities, _class)
+
+    # set up test cases
+    # 'setup' = arguments for `_get_validations`
+    # 'output' = expected return from `_get_validations`
+    # 'raises' = if `_get_validations` raises an Exception
+    # 'warns' = if `_get_validations` issues a warning
+    @pytest.mark.parametrize(
+        "case",
+        [
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        "func, func_args, func_kwargs, validations, expected, _warns",
        [
```

With this restructuring, we don't have to use the grab-bag variable `case` anymore.  By now injecting specific test arguments `test_vq_method__get_validations` will be much more readable.

I can suggest this directly, but the updated test look should now be... (This does remove the cases that raise exceptions.)

```python
    @pytest.mark.parametrize(
        "func, func_args, func_kwargs, validations, expected, _warns",
        [
            (  # typical call...using 'can_be_negative'
                _foo,
                (5,),
                {},
                {"x": {"units": u.cm, "can_be_negative": False}},
                {
                    "x": {
                        **default_validations,
                        "units": [u.cm],
                        "can_be_negative": False,
                    },
                },
                does_not_warn,
            ),
            (  # typical call...using 'none_shall_pass'
                _foo,
                (5,),
                {},
                {"x": {"units": u.cm, "none_shall_pass": True}},
                {
                    "x": {
                        **default_validations,
                        "units": [u.cm],
                        "none_shall_pass": True,
                    }
                },
                does_not_warn,
            ),
            (  # call w/o value validations
                _foo,
                (5,),
                {},
                {"x": {"units": u.cm}},
                {"x": {**default_validations, "units": [u.cm]}},
                does_not_warn,
            ),
            (  # 'none_shall_pass' defined w/ validations
                _foo,
                (5,),
                {},
                {"x": {"units": [u.cm, None]}},
                {
                    "x": {
                        **default_validations,
                        "units": [u.cm],
                        "none_shall_pass": True,
                    }
                },
                does_not_warn,
            ),
            (  # units are defined via function annotations
                _foo_anno,
                (5,),
                {},
                {},
                {"x": {**default_validations, "units": [u.cm]}},
                does_not_warn,
            ),
            (  # define 'validations_on_return'
                _foo,
                (5,),
                {},
                {"validations_on_return": {"units": [u.cm, None]}},
                {
                    "validations_on_return": {
                        **default_validations,
                        "units": [u.cm],
                        "none_shall_pass": True,
                    }
                },
                does_not_warn,
            ),
            (  # define both validations on args and validations_on_return
                _foo,
                (5,),
                {},
                {
                    "x": {"units": [u.cm], "none_shall_pass": False},
                    "validations_on_return": {"units": [u.cm], "can_be_zero": False},
                },
                {
                    "x": {
                        **default_validations,
                        "units": [u.cm],
                        "none_shall_pass": False,
                    },
                    "validations_on_return": {
                        **default_validations,
                        "units": [u.cm],
                        "can_be_zero": False,
                    },
                },
                does_not_warn,
            ),
        ],
    )
```